Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -647,6 +654,29 @@ private async Task<BoolResult> RestoreCheckpointStateAsync(OperationContext cont
                         Task.Run(() => ReconcileAsync(context), context.Token).FireAndForget(context);
                     }
                 }
+
+                // Make sure to only have one thread doing proactive replication, in case last proactive replication is still happening.
+                if (_configuration.EnableProactiveReplication && Interlocked.CompareExchange(ref _replicating, 1, 0) == 0)

```

Comment:
If previous proactive replication task is still running, this skips current cycle and lets old one run. It would be better to stop old task and start a new one, since we now have better information than old cycle.